GenderPayGap.WebUI/BackgroundJobs/ScheduledJobs/PurgeUsersJob.cs

```diff
@@ -27,63 +24,38 @@ public PurgeUsersJob(IDataRepository dataRepository, AuditLogger auditLogger)
-        public async Task PurgeUsers()
+        public void PurgeUsers()
         {
-            string runId = JobHelpers.CreateRunId();
-            DateTime startTime = VirtualDateTime.Now;
-            JobHelpers.LogFunctionStart(runId, nameof(PurgeUsers), startTime);
+            JobHelpers.RunAndLogJob(PurgeUsersAction, nameof(PurgeUsers));
+        }
 
+        private void PurgeUsersAction()
+        {
             DateTime deadline = VirtualDateTime.Now.AddDays(0 - Global.EmailVerificationExpiryDays);
             DateTime pinExpiryDate = VirtualDateTime.Now.AddDays(0 - Global.PinInPostExpiryDays);
 
             List<User> users = dataRepository.GetAll<User>()
+                .Where(u => u.Status == UserStatuses.New || u.Status == UserStatuses.Active)
                 .Where(u => u.EmailVerifiedDate == null)
                 .Where(u => u.EmailVerifySendDate == null || u.EmailVerifySendDate.Value < deadline)
                 .Include(u => u.UserOrganisations)
                 .ToList();
 
             foreach (User user in users)
             {
-                PurgeUser(user, pinExpiryDate, runId, startTime);
-            }
-
-            JobHelpers.LogFunctionEnd(runId, nameof(PurgeUsers), startTime);
-        }
-
-        private void PurgeUser(User user, DateTime pinExpiryDate, string runId, DateTime startTime)
-        {
-            try
-            {
-                auditLogger.AuditChangeToUser(
-                    AuditedAction.PurgeUser,
-                    user,
-                    new {user.UserId, user.EmailAddress, user.JobTitle, user.Fullname},
-                    null);
-
-                DeleteUserAndAuditLogs(user);
-            }
-            catch (Exception ex)
-            {
-                JobHelpers.LogFunctionError(runId, nameof(PurgeUsers), startTime, ex);
+                PurgeUser(user, pinExpiryDate);
             }
         }
 
-        private static void DeleteUserAndAuditLogs(User user)
+        private void PurgeUser(User user, DateTime pinExpiryDate)
         {
-            var dataRepository = MvcApplication.ContainerIoC.Resolve<IDataRepository>();
-
-            List<AuditLog> auditLogs = dataRepository
-                .GetAll<AuditLog>()
-                .Where(log => log.ImpersonatedUser.UserId == user.UserId)
-                .Where(log => log.Action == AuditedAction.AdminResendVerificationEmail)
-                .ToList();
-
-            foreach (AuditLog log in auditLogs)
-            {
-                dataRepository.Delete(log);
-            }
+            auditLogger.AuditChangeToUser(
+                AuditedAction.PurgeUser,
+                user,
+                new {user.UserId, user.EmailAddress, user.JobTitle, user.Fullname},
+                null);
 
-            dataRepository.Delete(user);
+            user.SetStatus(UserStatuses.Retired, user, "User retired by PurgeUserJob");

```

Comment:
Is `retired` for users the same as `deleted` for organisations? Or do they mean different things?

If they mean the same thing, would it be clearer to use the same status phrasing?

Reply:
Organisations have both Retired and Deleted:
* Retired is if they've stopped trading, but we still want to be able to see their old returns
* Deleted is if they were registered by mistake (e.g. if someone registered the same organisation twice)

Users just have Retired, which can be for:
* when they close their account
* if they create an account, but don't verify their email address in time

I think it's a good question as to whether Users-Retired is closer to Retired or Deleted, or whether we should have two statuses for the two ways in which they can become retired.
I don't think there's a clear answer either way!